File: src/codes/duplicate-logic/ts/uncode.ts
class PdfExporter {
  export(data: string) {
    console.log("開始ログ");
    console.log(`PDF出力: ${data}`);
    console.log("完了ログ");
  }
}

class CsvExporter {
  export(data: string) {
    console.log("開始ログ");
    console.log(`CSV出力: ${data}`);
    console.log("完了ログ");
  }
}

class XmlExporter {
  export(data: string) {
    console.log("開始ログ");
    console.log(`XML出力: ${data}`);
    console.log("完了ログ");
  }
}

File: src/codes/duplicate-logic/py/uncode.py
class PdfExporter:
    def export(self, data: str):
        print("開始ログ")
        print(f"PDF出力: {data}")
        print("完了ログ")

class CsvExporter:
    def export(self, data: str):
        print("開始ログ")
        print(f"CSV出力: {data}")
        print("完了ログ")

class XmlExporter:
    def export(self, data: str):
        print("開始ログ")
        print(f"XML出力: {data}")
        print("完了ログ")
